### ISSUE TITLE
[A11y]: Accessibility issues in Controls: missing labels, incorrect ARIA use, non-semantic interactive elements, row announcement pollution

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#32250]
### Describe the bug

There are a few accessibility issues in the controls rendering in Storybook that cause Axe and other a11y tools to raise warnings or errors. These issues can hinder usability for screen reader users and reduce compliance with accessibility standards such as WCAG.

### 1. Incorrect use of aria-readonly on radio button groups
Radio button groups in the ArgsTable are wrapped in a div with `aria-readonly="false"`.

This is problematic because aria-readonly is not valid here, and is intended for editable elements like input, textarea, or roles like textbox or with radio group with the correct role added.

Recommendation: I think the `aria-readonly="false"` should be removed or a `role="radiogroup"` should be added to that same div.

### 2. Missing labels for select and input controls
The select and input elements rendered in the ArgsTable controls do not have associated `<label>` elements.

This causes Axe to flag them as inaccessible since screen readers cannot determine their purpose.

### Reproduction link

https://next.design-system.post.ch/

### Reproduction steps

Issue 1 (radio buttons): Go to `Components > Button` and the issue is visible on the first table of the page
Issue 2 (missing labels): Go to `Components > Button` and the issue is visible on the first table of the page
Issue 3 (expandable content): Go to `Components > Accordion` and the issue is visible on the second table of the page

### System

```bash
Storybook v8.4.7
```
